myenumerables.rb

```diff
@@ -0,0 +1,158 @@
+module Enumerable
+  def my_each
+    index = 0
+    array = to_a
+    if block_given?
+      for item in 0..array.length - 1
+        yield(self[item])
+      end
+    else
+      while index < array.size
+        return to_enum(:my_each) unless block_given?
+
+        yield self[index]
+        index += 1
+        return self
+      end
+    end
+  end
+
+  def my_each_with_index(index = 0)
+    while index < size
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield self[index], index
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) == true }
+    result
+  end
+
+  def my_all?(query = nil)
+    if query.class == Class
+      my_each do |idx|
+        return false unless idx.class == query
+      end
+    elsif query.class == String or query.class == Integer
+      my_each do |idx|
+        return false unless idx == query
+      end
+    elsif query.class == Regexp
+      my_each do |idx|
+        return false unless idx.match(query)
+      end
+    elsif block_given?
+      my_each do |idx|
+        return false unless yield(idx)
+      end
+    elsif query.nil?
+      my_each do |idx|
+        return false if idx === false
+      end
+    else
+      my_each do |idx|
+        return false unless !idx.nil?
+      end
+    end
+    true
+  end
+
+  def my_any?(*query)
+    if block_given?
+      length.times { |item| return true unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return true unless item == query }
+    elsif query.class == Class
+      length.times { |item| return true unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return true unless item == query }
+    elsif query.class == Proc
+      length.times { |item| return true unless yield(item) }
+    else
+      length.times { |item| return true unless item }
+    end
+    false
+  end
+
+  def my_none?(*query)
+    if block_given?
+      length.times { |item| return false unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return false unless item == query }
+    elsif query.class == Class
+      length.times { |item| return false unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return false unless item == query }
+    else
+      length.times { |item| return false unless item }
+    end
+    true
+  end
```

Comment:
```rb
2.6.3 :034 > array
 => [3, 8, 5, 6, 1, 5, 8, 3, 7, 7, 6, 0, 3, 6, 7, 7, 2, 2, 1, 1, 4, 3, 5, 1, 4, 5, 2, 4, 8, 7, 2, 7, 8, 2, 2, 7, 4, 0, 6, 5, 7, 7, 6, 4, 1, 8, 5, 5, 7, 0, 3, 3, 1, 0, 1, 5, 5, 7, 3, 8, 5, 2, 1, 5, 3, 8, 6, 3, 5, 5, 6, 4, 0, 7, 5, 7, 7, 7, 7, 8, 1, 6, 8, 3, 1, 6, 5, 6, 5, 7, 4, 1, 8, 8, 6, 1, 1, 0, 8, 5] 
2.6.3 :035 > true_block = proc { |num| num > 9 }
 => #<Proc:0x000000000212c3a0@(irb):35> 
2.6.3 :036 > array.none?(&true_block)
 => true 
2.6.3 :037 > array.my_none?(&true_block)
 => false 
2.6.3 :038 >
```

Reply:
```rb
2.6.3 :040 > true_array = [nil, false, true, []]
 => [nil, false, true, []] 
2.6.3 :041 > true_array.none?
 => false 
2.6.3 :042 > true_array.my_none?
 => true 
2.6.3 :043 >
```

Reply:
that error https://github.com/rootDEV2990/ruby_enumerables_rewrites/pull/1#discussion_r429650092  is still there

Reply:
that error https://github.com/rootDEV2990/ruby_enumerables_rewrites/pull/1#discussion_r429650171 still persist

Reply:
```rb
2.6.3 :063 > ws[0] = 5
 => 5 
2.6.3 :064 > ws.none?(5)
 => false 
2.6.3 :065 > ws.my_none?(5)
 => true 
2.6.3 :066 > ws
 => [5, "door", "rod", "blade"] 
2.6.3 :067 > 
```